### DISCUSSION TITLE
Playing audio with the RaspBerry Pi Pico

### DISCUSSION THREAD

**Discussion by greenreader9** [micropython/micropython#10361] [Using MicroPython]
Hello!

Hopefully I picked the correct category, if not, my apologies.

I am trying to get a speaker connected to the Raspberry pi Pico W using MicroPython, but without much success. 

I started with Bluetooth, but quickly ruled that out as the Pico W does not support it (At least without a lot of hacking) yet. 

So wired then. My idea was to take a TS 3.5mm cable, cut it, and somehow connect it to the Pi. Since the speaker on the other end has a battery, signal strength should not be an issue, right? (I am a website developer, not an audio engineer). I really want to do this in MicroPython, since the rest of my code works well and I don't want to change it. 

I came across this respiratory (https://github.com/danjperron/PicoAudioPWM), and it looks like it might work, but the first 4 paragraphs are totally confusing. (Would I be using a TRS 3.5mm cable with this)?

In conclusion, I'm really just trying to get some audio to play over a 3.5mm cable with the Pi Pico. Would the above linked GH respiratory be the way to go, and how would I wire up my speaker with it?

Thanks for any assistance you can offer!

**Comment by jimmo** [comment]
@greenreader9 Sorry about the  delay responding...

> Since the speaker on the other end has a battery, signal strength should not be an issue, right?

This is an important detail to clarify. What exactly is the "speaker". Sounds like you're actually driving something that expects a "line out" signal?

> I came across this respiratory (https://github.com/danjperron/PicoAudioPWM), and it looks like it might work, but the first 4 paragraphs are totally confusing. (Would I be using a TRS 3.5mm cable with this)?

Yes. This could be made to work. You will need to do some research to figure out exactly what voltages and therefore which size resistor to use.

If you need some help explaining the paragraphs from that repo then feel free to ask questions here!

**Reply by greenreader9** [reply]
Thanks for the reply!

The speaker is a small Bluetooth battery-powered rechargeable  speaker with a 3.5mm line-in jack that I was given as a freebie a few years back (I want it to do something other then collect dust). 

When you say that I will have to do some research, are you talking about researching what the speaker can handle (I thought that was all standardized, but I could be wrong), or am I researching the voltage necessary to be outputted by the Pico in order for the speaker to read it?

Thanks!

**Comment by peterhinch** [comment]
The solution depends on the type of sounds you want to produce. PWM is OK for synthetic clicks, beeps and honks or simple monophonic tunes.

If you want to play music, you'll need something better which will involve buying hardware. One approach is to use [I2S](https://docs.micropython.org/en/latest/library/machine.I2S.html#class-i2s-inter-ic-sound-bus-protocol) with a cheap UDA1334 DAC board from eBay. This will play WAV files. For MP3 or FLAC you'll need a [VS1053](https://github.com/peterhinch/micropython-vs1053).

**Reply by greenreader9** [reply]
Since the audio I am hoping to play is a persons voice, I am assuming I will need to encode it as a WAV file and use the I2S protocol then?

**Reply by peterhinch** [reply]
That's certainly the route to high audio quality.
